# 14 · GSE232381 · bulk_RNA_seq · WGCNA modules (exploratory)

Reads `wgcna_input.rds`. Writes `modules.rds` and the end product `module_genes.csv` in
`data/run_artifacts/GSE232381/`.

Settings as for GSE65391 (notebook 05): signed network and TOM, minModuleSize 30, mergeCutHeight 0.25.
There are more genes than fit one block, so `maxBlockSize = 5000`: WGCNA pre-clusters the genes into
blocks and builds modules within each block.

## Load WGCNA and the input from notebook 13

In [1]:
if (!requireNamespace("WGCNA", quietly = TRUE)) install.packages("WGCNA", repos = "https://cloud.r-project.org")
suppressMessages(library(WGCNA))
source("../src/paths.R")
w <- readRDS(art("GSE232381", "wgcna_input.rds"))
c(samples = nrow(w$datExpr), genes = ncol(w$datExpr), power = w$power)

samples   genes   power 
     16   16605      16

**Explanation**
- As in notebook 05, with the input of notebook 13.

## Find the modules

In [2]:
set.seed(SEED)
net <- blockwiseModules(w$datExpr, power = w$power, networkType = "signed", TOMType = "signed",
                        minModuleSize = 30, mergeCutHeight = 0.25, maxBlockSize = 5000,
                        numericLabels = FALSE, pamRespectsDendro = FALSE, verbose = 0)
sizes <- sort(table(net$colors), decreasing = TRUE)
c(modules = sum(names(sizes) != "grey"), genes_in_modules = sum(sizes[names(sizes) != "grey"]), grey = unname(sizes["grey"]))
sizes

modules genes_in_modules             grey 
              39            16500              105


     turquoise           blue          brown         yellow          green 
          4030           3207           1737            713            660 
           red          black           pink        magenta         purple 
           594            473            457            442            359 
   greenyellow            tan         salmon           cyan   midnightblue 
           326            280            276            261            257 
     lightcyan         grey60     lightgreen    lightyellow      royalblue 
           238            207            182            178            152 
       darkred      darkgreen  darkturquoise       darkgrey           grey 
           142            134            130            125            105 
        orange     darkorange          white        skyblue    saddlebrown 
            99             83             80             74             67 
 paleturquoise      steelblue         violet darkolivegreen    darkmagenta 
           

**Explanation**
- `blockwiseModules()` as in notebook 05, with one difference: `maxBlockSize = 5000`. With more genes than
  that, WGCNA first splits the genes into blocks of at most 5,000 related genes and finds modules within each
  block. This limits the memory needed.
- The counts of modules, genes in modules and grey genes as in notebook 05.

**Result.** 39 modules holding 16,500 of 16,605 genes; 105 are grey. With 16 samples, correlations between
unrelated genes are large by chance, so almost every gene joins a module. This is what the FAQ's warning
about small samples predicts.

## Module membership and hub genes

In [3]:
ME  <- orderMEs(net$MEs)
kME <- cor(w$datExpr, ME)
modules <- setdiff(names(sizes), "grey")
hubs <- lapply(setNames(modules, modules), function(mo) {
  g <- names(net$colors)[net$colors == mo]
  names(sort(kME[g, paste0("ME", mo)], decreasing = TRUE))
})
t(sapply(hubs, function(h) paste(head(h, 8), collapse = ", ")))

turquoise,blue,brown,yellow,green,red,black,pink,magenta,purple,⋯,paleturquoise,steelblue,violet,darkolivegreen,darkmagenta,sienna3,skyblue3,yellowgreen,plum1,orangered4
"TADA1, ZNF675, TRAT1, TMEM209, LYRM7, ZBTB35, N4BP2, SRFBP1","ATP6AP1, PTTG1IP, FURIN, STXBP2, UBAP1, HCLS1, VSIR, FCER1G","RGL2, SMIM48, RCN3, SAP25, ABCA7, UNKL, LPAR2, FAM219A","MRPS7, ILF2, MRPL14, NDUFAB1, PHB1, UFC1, NHP2, GATD3","LOC102723665, PGM5P2, TTC12, ALS2CL, MRPS31P5, GVQW3, RRN3P1, LINC00665","ZDHHC20, RAB8B, TBC1D23, UBLCP1, MTMR6, RC3H2, POC1B, EFR3A","LRMDA, FUCA2, COMMD9, CCDC22, ATP13A2, ARMH3, SNAPC2, RPS6KA4","NDUFA11, PSMA7, DUSP23, AIP, PSMD8, EIF6, SNRPB, EDF1","NIN, MAP3K2, RNF111, GSK3B, RYBP, KBTBD2, NUP58, KDM5A","NLN, EMC1, CCDC88A, SHTN1, HNMT, NARS1, TANGO6, ATRN",⋯,"METTL25, SEC22B3P, SEC22B2P, MTMR7, PIGA, PAXBP1-AS1, CASK, EPM2A-DT","HIF1A-AS1, DPYD-AS1, SPDYE10, ATF7-AS1, LINC00260, LOC107986710, PEAK1, MIR4315-1","RNA5S14, RNA5S13, RNA5S17, RNA5S1, RNA5S6, RNA5S15, RNA5S4, RNA5S11","CEACAM6, CEACAM8, PRTN3, MMP8, DEFA4, ELANE, AZU1, LINC02009","H2AC17, H2BC17, H2BC11, H2BC9, PDGFRA, EXO5, H2AC8, LOC102724765","CYP4F22, MS4A2, CPA3, CXCL10, IDO1, APOL6, SAMD9, EFNB1","SRCAP, GALNT6, EIF4G1, DYNC1H1, FBXL18, SAP130, SLC38A12, OSBP","ATF7, DOP1B, ARHGEF7, TIRAP, PPARGC1B, MIR6859-4, PLD1, VPS53","KCNH3, NCBP3, USP48, CCDC125, PDCD6IP, PIAS2, TCP11L2, CPM","HECTD1, MED14, DDI2, YLPM1, SENP5, RSPRY1, USP24, USP47"


**Explanation**
- kME and the hub genes of each module as in notebook 05.

## Which modules hold the interferon genes?

In [4]:
ifn6 <- read_gene_set("ifn-type1-6.txt")
data.frame(gene = ifn6, module = net$colors[ifn6], row.names = NULL)

gene,module
<chr>,<chr>
IFI27,darkorange
IFI44L,darkred
IFIT1,darkred
ISG15,darkred
RSAD2,darkred
SIGLEC1,darkred


**Explanation**
- As in notebook 05: the module of each of the six interferon-score genes.

**Result.** Three modules read clearly by their hub genes: darkred (OAS2, XAF1, MX1, IFIT3, OAS3:
interferon), darkolivegreen (CEACAM6, CEACAM8, PRTN3, MMP8, DEFA4, ELANE: neutrophil granule) and greenyellow
(8 of its 10 hub genes are IGHV, IGKV or IGLV genes: immunoglobulin, plasma cell). Five of the six
interferon-score genes are in darkred; IFI27 is in darkorange.

## Save the modules

In [5]:
in_module <- names(net$colors)[net$colors != "grey"]
module_genes <- data.frame(gene = in_module, module = net$colors[in_module],
                           kME = kME[cbind(in_module, paste0("ME", net$colors[in_module]))], row.names = NULL)
write.csv(module_genes, art("GSE232381", "module_genes.csv"), row.names = FALSE)
saveRDS(list(colors = net$colors, MEs = ME, kME = kME, hubs = hubs), art("GSE232381", "modules.rds"))

**Explanation**
- As in notebook 05. `module_genes.csv` lists this study's own modules; they are not the array modules.